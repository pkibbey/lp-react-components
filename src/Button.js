import React from 'react'
import { Button as RebassButton } from 'rebass'
import Loader from './Loader'
import theme from './assets/theme'

const getStyles = (variant) => {
  switch (variant) {
    case 'secondary':
      return {
        fontSize: theme.fontSizes[0],
        fontFamily: '"Roboto", sans-serif',
        fontWeight: 700,
        padding: '0 24px',
        height: '32px',
        borderRadius: 32,
        cursor: 'pointer',
        backgroundColor: theme.colors.lighterGray,
        color: 'darkGray',
        WebkitFontSmoothing: 'antialiased',
        textDecoration: 'none',
        transition: 'all 2s',
        '&:hover': {
          backgroundColor: theme.colors.lightGray,
          color: 'darkerGray'
        }
      }
    default:
      return {
        fontSize: theme.fontSizes[2],
        fontFamily: '"Roboto", sans-serif',
        fontWeight: 700,
        height: '62px',
        paddingTop: 0,
        paddingBottom: 0,
        borderRadius: 32,
        letterSpacing: '0.008em',
        cursor: 'pointer',
        backgroundColor: theme.colors.electricBlue,
        color: 'white',
        WebkitFontSmoothing: 'antialiased',
        maxWidth: '100%',
        whiteSpace: 'nowrap',
        width: '100%',
        transition: 'all 2s',
        '&:hover': {
          backgroundColor: theme.colors.electricBlueHover
        }
      }
  }
}

const Button = ({ isLoading, handleClick, label, variant, ...otherProps }) => (
  <RebassButton onClick={handleClick} sx={getStyles(variant)} {...otherProps}>
    {isLoading ? (
      <Loader size={variant !== 'default' ? 'small' : 'default'} />
    ) : (
      label
    )}
  </RebassButton>
)

export default Button
