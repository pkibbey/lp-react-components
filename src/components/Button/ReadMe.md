Default Button:

```js
<Button label="Submit" />
```

Loading Button:

```js
<Button label="Label not shown" isLoading={true} />
```

Secondary Button:

```js
<Button variant="secondary" label="Options" />
```
