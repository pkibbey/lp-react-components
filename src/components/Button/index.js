import React from 'react'
import { Button as RebassButton } from 'rebass'
import Loader from '../Loader/'
import PropTypes from 'prop-types'
import ThemeWrapper from '../ThemeWrapper'

/**
 * Used to provide call-to-action elements
 */
const Button = ({
  isLoading,
  handleClick,
  label,
  variant,
  isFullWidth,
  ...otherProps
}) => (
  <ThemeWrapper>
    <RebassButton
      onClick={handleClick}
      variant={variant}
      sx={{ width: isFullWidth ? '100%' : 'auto' }}
      {...otherProps}
    >
      {isLoading ? (
        <Loader
          size={variant === 'primary' ? 'default' : 'small'}
          variant={variant === 'secondary' ? 'dark' : 'light'}
        />
      ) : (
        label
      )}
    </RebassButton>
  </ThemeWrapper>
)

Button.propTypes = {
  /** The button label */
  label: PropTypes.string,
  /** A callback to fire when the button is clicked */
  handleClick: PropTypes.func,
  /** A variant for the button */
  variant: PropTypes.oneOf(['primary', 'secondary']),
  /** if this is true, it shows a loading animation in the button */
  isLoading: PropTypes.bool,
  /** if this is true, it shows a loading animation in the button */
  isFullWidth: PropTypes.bool
}

Button.defaultProps = {
  label: '',
  handleClick: undefined,
  variant: 'primary',
  isLoading: false,
  isFullWidth: false
}

export default Button
