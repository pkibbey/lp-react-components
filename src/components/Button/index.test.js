import React from 'react'
import renderer from 'react-test-renderer'
import Button from './'

it('renders a default button', () => {
  const tree = renderer.create(<Button />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders a loading button', () => {
  const tree = renderer.create(<Button isLoading />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders a wide button', () => {
  const tree = renderer.create(<Button isFullWidth />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders a secondary button', () => {
  const tree = renderer
    .create(<Button isLoading variant='secondary' />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
