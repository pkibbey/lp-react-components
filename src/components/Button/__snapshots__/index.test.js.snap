// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a default button 1`] = `
<button
  className="css-k73zv0"
>
  
</button>
`;

exports[`renders a loading button 1`] = `
<button
  className="css-k73zv0"
>
  <div
    className="react-spinner-material"
    data-testid="loader"
    style={
      Object {
        "borderColor": "white",
        "borderWidth": 2,
        "height": 24,
        "width": 24,
      }
    }
    type="Oval"
  />
</button>
`;

exports[`renders a secondary button 1`] = `
<button
  className="css-kl71ql"
>
  <div
    className="react-spinner-material"
    data-testid="loader"
    style={
      Object {
        "borderColor": "#675D55",
        "borderWidth": 2,
        "height": 18,
        "width": 18,
      }
    }
    type="Oval"
  />
</button>
`;

exports[`renders a wide button 1`] = `
<button
  className="css-ipe9fh"
>
  
</button>
`;
